samples/data-sealing/common/dispatcher.cpp

```diff
@@ -2,382 +2,113 @@
-    ret = cipher_data(
-        ENCRYPT_OPERATION,
-        m_data,
-        (unsigned int)m_data_size,
-        seal_key,
-        (unsigned int)seal_key_size,
-        iv,
-        m_sealed_data->encrypted_data);
-    if (ret != 0)
-    {
-        TRACE_ENCLAVE("cipher_data failed with %d\n", ret);
-        goto exit;
-    }
-
-    // On the return from above cipher_data, the iv value was updated ,
-    // initialize to the original value before using it in sign_sealed_data
-    memcpy(iv, m_sealed_data->iv, IV_SIZE);
-
-    // generate signature by signing the hash of the sealed data with the seal
-    // key
-    ret = sign_sealed_data(
-        m_sealed_data,
-        seal_key,
-        (unsigned int)seal_key_size,
-        m_sealed_data->signature);
-    if (ret != 0)
-    {
-        TRACE_ENCLAVE("sign_sealed_data %d\n", ret);
+        TRACE_ENCLAVE("blob_size is too large to fit into an unsigned int");
+        ret = OE_OUT_OF_MEMORY;
         goto exit;
     }
 
     temp_sealed_data =
-        (sealed_data_t*)oe_host_malloc(m_sealed_data->total_size);
+        (sealed_data_t*)oe_host_malloc(sizeof(*temp_sealed_data) + blob_size);
     if (temp_sealed_data == NULL)
     {
-        result = OE_OUT_OF_MEMORY;
+        ret = OE_OUT_OF_MEMORY;
         goto exit;
     }
-    memcpy(temp_sealed_data, m_sealed_data, m_sealed_data->total_size);
-
-    *sealed_data_size = m_sealed_data->total_size;
-    *sealed_data = temp_sealed_data;
-exit:
-
-    if (m_data)
-    {
-        free(m_data);
-        m_data = NULL;
-    }
 
-    if (seal_key)
-        free(seal_key);
+    memset(temp_sealed_data, 0, sizeof(*temp_sealed_data));
+    memcpy(temp_sealed_data->opt_msg, opt_msg, opt_msg_len);
+    temp_sealed_data->sealed_blob_len = blob_size;
+    memcpy(temp_sealed_data + 1, blob, blob_size);
 
-    if (m_sealed_data)
-    {
-        free(m_sealed_data);
-        m_sealed_data = NULL;
-    }
+    *sealed_data = temp_sealed_data;
+    *sealed_data_size = sizeof(*temp_sealed_data) + blob_size;
 
-    if (ret)
-        result = OE_FAILURE;
-    return (int)result;
+exit:
+    oe_free(blob);
+    return ret;
 }
 
 int ecall_dispatcher::unseal_data(
-    sealed_data_t* sealed_data,
+    const sealed_data_t* sealed_data,
     size_t sealed_data_size,
     unsigned char** data,
     size_t* data_size)
 {
-    oe_result_t result = OE_OK;
-    unsigned char iv[IV_SIZE];
-    unsigned char signature[SIGNATURE_LEN];
-    uint8_t* seal_key = NULL;
-    size_t seal_key_size = 0;
-    uint8_t* key_info = NULL;
-    size_t key_info_size = 0;
-
-    unsigned char* data_buf = NULL;
-    int ret = 0;
+    uint8_t* temp_data;
 
-    key_info = sealed_data->encrypted_data + sealed_data->encrypted_data_len;
-    key_info_size = sealed_data->key_info_size;
-
-    m_sealed_data = sealed_data;
-    *data_size = 0;
-    *data = NULL;
-
-    if (sealed_data_size != sealed_data->total_size)
+    if (sealed_data_size != sealed_data->sealed_blob_len + sizeof(*sealed_data))
```

Comment:
"sealed_blob_len" should be changed to "sealed_blob_size" to avoid abbreviations